src/org/labkey/test/components/ui/grids/EditableGrid.java

```diff
@@ -351,10 +365,26 @@ public void setCellValue(int row, String columnName, Object value)
             // Treat the object being sent in as a string.
             // Get the inputCell enter the text and then make the inputCell go away (hit RETURN).
 
-            WebElement inputCell = elementCache().inputCell();
-            inputCell.clear();
+            if (useDatePicker)
+            {
+                ReactDatePicker datePicker = elementCache().datePicker();
+                datePicker.set("", false);
+                String[] dateParts = getParsedDateParts((String) value);
+                datePicker.set(dateParts[0], false); // use keyboard input to set year and month
+                elementCache().datePickerDateCell(dateParts[1]).click(); // use calendar ui to select day
+                if (!StringUtils.isEmpty(dateParts[2])) // use timepicker to select time
+                    datePicker.clickTime(dateParts[2]);
+
+                WebElement inputCell = elementCache().inputCell();
+                inputCell.sendKeys(Keys.RETURN);
```

Comment:
Move the calendar and day pickers into `ReactDatePicker`:
```suggestion
                ReactDatePicker datePicker = elementCache().datePicker();
                String[] dateParts = getParsedDateParts((String) value);
                datePicker.setWithPopup(dateParts[0], dateParts[1], dateParts[2]);
```